File: create_session.py
"""
Создаёт StringSession ЛОКАЛЬНО на вашем ПК.
Запуск: python create_session.py

Скопируйте полученную строку в Dokploy → Environment → TELETHON_SESSION
"""

import asyncio
from telethon import TelegramClient
from telethon.sessions import StringSession
from telethon.errors import SessionPasswordNeededError

API_ID = 33112824
API_HASH = "7526146bb71fe8ca2fcf1b353f34425e"


async def main():
    print("=" * 50)
    print("🔐 Создание Telethon StringSession")
    print("=" * 50)
    print()

    client = TelegramClient(StringSession(), API_ID, API_HASH)
    await client.connect()

    phone = input("Введите номер телефона (+79XXXXXXXXX): ").strip()

    sent = await client.send_code_request(phone)
    print()
    print("📱 Код отправлен в Telegram!")
    print()

    code = input("Введите код из Telegram: ").strip()

    try:
        await client.sign_in(phone, code, phone_code_hash=sent.phone_code_hash)
    except SessionPasswordNeededError:
        print()
        print("🔒 Двухфакторная аутентификация включена.")
        password = input("Введите пароль 2FA: ").strip()
        await client.sign_in(password=password)

    me = await client.get_me()
    session_string = client.session.save()

    print()
    print(f"✅ Авторизация успешна! Вы: {me.first_name}")
    print()
    print("=" * 50)
    print("📋 СКОПИРУЙТЕ ЭТУ СТРОКУ:")
    print("=" * 50)
    print()
    print(session_string)
    print()
    print("=" * 50)
    print()
    print("Вставьте её в Dokploy → Environment:")
    print("  TELETHON_SESSION=<строка выше>")
    print()
    print("Затем нажмите Redeploy")

    await client.disconnect()


if __name__ == "__main__":
    asyncio.run(main())

File: create_session.mjs
/**
 * Создаёт Telethon-совместимую StringSession с помощью GramJS (JS-порт Telethon).
 * Запуск: node create_session.mjs
 */

import { TelegramClient } from "telegram";
import { StringSession } from "telegram/sessions/index.js";
import readline from "readline";

const API_ID = 33112824;
const API_HASH = "7526146bb71fe8ca2fcf1b353f34425e";

const rl = readline.createInterface({ input: process.stdin, output: process.stdout });
const ask = (q) => new Promise((r) => rl.question(q, r));

async function main() {
    console.log("=".repeat(50));
    console.log("🔐 Создание Telethon StringSession (Node.js)");
    console.log("=".repeat(50));
    console.log();

    const session = new StringSession("");
    const client = new TelegramClient(session, API_ID, API_HASH, {
        connectionRetries: 3,
    });

    await client.start({
        phoneNumber: async () => await ask("Введите номер телефона (+79XXXXXXXXX): "),
        phoneCode: async () => await ask("Введите код из Telegram: "),
        password: async () => await ask("Введите пароль 2FA: "),
        onError: (err) => console.error("Ошибка:", err.message),
    });

    const me = await client.getMe();
    const sessionStr = client.session.save();

    console.log();
    console.log(`✅ Авторизация успешна! Вы: ${me.firstName}`);
    console.log();
    console.log("=".repeat(50));
    console.log("📋 СКОПИРУЙТЕ ЭТУ СТРОКУ:");
    console.log("=".repeat(50));
    console.log();
    console.log(sessionStr);
    console.log();
    console.log("=".repeat(50));
    console.log();
    console.log("Вставьте в Dokploy → Environment:");
    console.log("  TELETHON_SESSION=<строка выше>");
    console.log();
    console.log("Затем нажмите Redeploy");

    await client.disconnect();
    rl.close();
}

main().catch(console.error);
